### DISCUSSION TITLE
Callback to Reveal Focus

### DISCUSSION THREAD

**Discussion by LuanRoger** [bdlukaa/fluent_ui#166] [Q&A]
It would help a lot if there is a way to call a function when a widget gets focus by **Reveal Focus**.
What I think:
```dart
Focus(
  onGetFocus: () { print("This button is focused") },
  child: Button(
    text: Text('Button'),
    onPressed: () {},
  )
)
```

**Comment by bdlukaa** [comment]
Hello! The `Focus` widget is provided by the Flutter framework itself. To know when a widget gets focus, you can use a `FocusNode`:

```dart
final focusNode = FocusNode();

@override
void initState() {
  super.initState();
  focusNode.addListener(() {
    if (focusNode.hasFocus) print('widget got focus');
  });
}

Focus(
  focusNode: focusNode:
),
```

You can also use `FocusableActionDetector`. It provides a nice dev experience with callbacks to focus:

```dart
FocusableActionDetector(
  onShowFocusHighlight: (v) { v ? print('get focus') : print('lost focus';) },
  
),
```

See https://api.flutter.dev/flutter/widgets/FocusableActionDetector-class.html

The `fluent_ui` library provides `HoverButton`, which can detect any state for any widget:

```dart
HoverButton(
  onPressed: () { print('pressed'); },
  builder: (BuildContext context, states) {
    return FocusBorder(
      focused: states.isFocused,
      child: ...,
    );
  },
)
```
